File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        sdict = dict()
        tdict = dict()
        ref = 0

        for i in range(len(s)):
            if s[i] not in sdict and t[i] not in tdict:
                sdict[s[i]] = ref
                tdict[t[i]] = ref
                ref += 1
            elif s[i] not in sdict or t[i] not in tdict: break
            else:
                if sdict[s[i]] != tdict[t[i]]: break
        else: return True
        return False

File: 0205-isomorphic-strings/0205-isomorphic-strings.java
import java.util.HashMap;

class Solution {
    public boolean isIsomorphic(String s, String t) {
        HashMap<Character, Integer> sMap = new HashMap<>();
        HashMap<Character, Integer> tMap = new HashMap<>();
        int ref = 0;

        for (int i = 0; i < s.length(); i++) {
            if (!sMap.containsKey(s.charAt(i)) && !tMap.containsKey(t.charAt(i))) {
                sMap.put(s.charAt(i), ref);
                tMap.put(t.charAt(i), ref);
                ref++;
            } else if (!sMap.containsKey(s.charAt(i)) || !tMap.containsKey(t.charAt(i))) {
                return false;
            } else {
                if (sMap.get(s.charAt(i)) != tMap.get(t.charAt(i))) {
                    return false;
                }
            }
        }
        return true;
    }
}